quinn/src/streams.rs

```diff
@@ -353,42 +353,38 @@ where
         buf: &mut ReadBuf<'_>,
     ) -> Poll<Result<(), ReadError>> {
         self.poll_read_generic(cx, |conn, stream| {
-            conn.inner
-                .read(stream, buf.remaining())
-                .map(|val| val.map(|chunk| buf.put_slice(&chunk)))
+            let mut chunks = conn.inner.read(stream, true)?;
```

Comment:
Just a confirmation question: `chunks` is the iterator here, which directly points to the assembler, right? So this won't require a new stream lookup on every read? If that's true, then I'm fine with it. However I would really like to avoid having to go through the whole stack for retrieving 1kB of data. hashmap lookups are fast - but not that fast :-)

Reply:
The current impl moves the stream state from the hash table into the iterator for easy access, then sticks it back when it's done, which avoids O(n) hash lookups. We could probably take a reference instead with sufficient refactoring to satisfy borrowck, but that should probably be left for future work.

Reply:
The refactoring would be tricky since, when the `Chunks` is dropped, it may want to call `Streams::stream_freed()`, which actually manipulates the `recv` hash map.